platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/CustomGeometrySource.java

```diff
@@ -62,14 +62,16 @@ public CustomGeometrySource(String id, GeometryTileProvider provider) {
   }
 
   /**
-   * Create a CustomGeometrySource with non-default CustomGeometrySourceOptions.
-   * <p>Supported options are minZoom, maxZoom, buffer, and tolerance.</p>
-   *  @param id       The source id.
+   * Create a CustomGeometrySource with non-default {@link CustomGeometrySourceOptions}.
+   * <p>
+   * Supported inherited {@link GeoJsonOptions} options are minZoom, maxZoom, buffer, and tolerance.
+   *
+   * @param id       The source id.
    * @param options  CustomGeometrySourceOptions.
    * @param provider The tile provider that returns geometry data for this source.
    */
   @UiThread
-  public CustomGeometrySource(String id, CustomGeometrySourceOptions options,
+  public CustomGeometrySource(String id, GeoJsonOptions options,
```

Comment:
I'm not sure I understand the typing problem being addressed here. The documentation is missing the `wrap` and `clip` options that are also supported. If those are supported, would it be better to keep the `CustomGeometrySourceOptions` type so that a developer knows the right type to instantiate to access those additional properties?

Reply:
The constructor accepts also `GeoJsonOptions` base type and to make this abstraction easier to implement, without casting, I think that we should accept the base type as the argument type. The `CustomGeometrySourceOptions`, which contatins those additional options, is referenced in the javadoc as well.

Would adding another constructor for the second type make it more readable?

Reply:
> Would adding another constructor for the extended type make it more readable?

That wouldn't really help, as it would only be used automatically in a few cases. One option is to remove the inheritance in `CustomGeometrySourceOptions` and have it only contain the supported methods, returning the correct object type.

Reply:
Sure @asheemmamoowala, this makes sense.